### PR TITLE
Revoke Reasons Pop Up

### PR DESCRIPTION
**Pull request by dc-andysign**
---
name: Revoke Reasons Pop Up
about: Pop up that includes a list of revocation reasons including others
---

## Summary

Simple form or pop-up after the revocation button is clicked 

This adds a separate pop-up component that contains an entire form.

This pull request is strictly related to the **JIRA** ticket **HYP-142** or parts of it.

---

## Motivation

To make sure the React GUI is as close as possible to the Figma design.

---

## Describe alternatives you've considered

To have everything into one component

---

## Additional context

OriginalFigmaScrShot

image

FirstSection

ExampleReasonsAttachmentJSOBblob

```json
{
	"dataError":{"discrepancies":false,"incorrect":true,"missing":false},
	"certMis":{"incorrect":false,"unverified":false,"claims":false},
	"nonCompliance":{"violation":false,"repeated":false},
	"otherReason":"other"
}
```

WithConfirmButtons

image

image

image

---

### CODE REVIEW

File: src/pages/components/RevokeActionsButton.js
Comment (rmlearney-digicatapult) [inline src/pages/components/RevokeActionsButton.js:11]: should this be in the ui component library?

File: src/pages/components/RevokeActionsButton.js
Comment (rmlearney-digicatapult) [inline src/pages/components/RevokeActionsButton.js:59]: We shouldn't mention 'blockchain' anywhere. Please change this to the simpler: 
"Click the submit button to revoke the certificate"

This also means we need an obvious way to cancel certification - can we have a cancel button next to the submit button, or a big `X` in the top right of the dialog?

File: src/pages/components/RevokeActionsButton.js
Comment (rmlearney-digicatapult) [reply src/pages/components/RevokeActionsButton.js:188]: Also displaying unnecessary data to user

**Review by rmlearney-digicatapult — COMMENTED** [review]
Some changes requested

File: src/pages/components/RevokeActionsButton.js
Comment (dc-andysign) [reply src/pages/components/RevokeActionsButton.js]: Indeed, I was thinking about that as well .. 

the problem is : 

I don't know what type UI element this is .. is this a background icon , is this an image for the beginning of a paragraph.. is this a text with a custom image bullet-point ? ( not to mention the time constrain )

I feel like this is more like a question for others ..

File: src/pages/components/RevokeActionsButton.js
Comment (dc-andysign) [reply src/pages/components/RevokeActionsButton.js]: image

Comment (dc-andysign) [reply src/pages/components/RevokeActionsButton.js]: Done

File: src/pages/components/RevokeActionsButton.js
Comment (dc-andysign) [reply src/pages/components/RevokeActionsButton.js:28]: Done

File: src/pages/CertificateViewer.js
Comment (mattdean-digicatapult) [inline src/pages/CertificateViewer.js:77]: please remove the comment

**Review by mattdean-digicatapult — COMMENTED** [review]
I've looked throug hand this seems good. I'd like to wait for Rob's approval as he's doing some manual acceptance testing. Once he approves I'm basically happy. There's one comment you should remove I think

File: src/pages/CertificateViewer.js
Comment (dc-andysign) [reply src/pages/CertificateViewer.js]: Doing it now ...

DONE!

**Comment by rmlearney-digicatapult** [comment]
Some slightly ugly UI elements:

This needs to be properly formatted within a box
Screenshot 2024-03-27 at 10 43 15

and the blue box around the black X - both should be white or more obvious, and try to resize so you don't do a quarter-scroll down to have to see the whole form

Screenshot 2024-03-27 at 10 42 51

**Comment by dc-andysign** [comment]
> Some slightly ugly UI elements:
> 
> This needs to be properly formatted within a box Screenshot 2024-03-27 at 10 43 15
> 
> and the blue box around the black X - both should be white or more obvious, and try to resize header so you don't do a quarter-scroll down to have to see the whole form
> 
> Screenshot 2024-03-27 at 10 42 51

Done

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**
